manifests/profile/moku.pp

```diff
@@ -0,0 +1,130 @@
+# Copyright (c) 2019 The Regents of the University of Michigan.
+# All Rights Reserved. Licensed according to the terms of the Revised
+# BSD License. See LICENSE.txt for details.
+
+# nebula::profile::moku
+#
+# @example
+#   include nebula::profile::moku
+class nebula::profile::moku (
+  String $init_directory = '/etc/moku/init',
+) {
+  lookup('nebula::named_instances').each |$name, $instance| {
+    if has_key($instance, 'users') {
+      $users = to_json($instance['users'])
+    } else {
+      $users = '[]'
+    }
+
+    if has_key($instance, 'subservices') {
+      $subservices = to_json($instance['subservices'])
+    } else {
+      $subservices = '[]'
+    }
+
+    concat_file { "${name} deploy init":
+      path   => "${init_directory}/${name}.json",
+      format => 'json-pretty',
+    }
+
+    Concat_fragment <<| target == "${name} deploy init" |>>
+
+    concat_fragment {
+      default:
+        target  => "${name} deploy init",
+        ;
+
+      "${name} deploy init instance.source.url":
+        content => "{\"instance\": {\"source\": {\"url\": \"${instance['source_url']}\"}}}",
```

Comment:
It might be easier to read/maintain (as would the others below) if this was specified as:

```
{instance => {source => {url => $instance['source_url] }}}.to_json
```